3party/jansson/CMakeLists.txt

```diff
@@ -4,14 +4,4 @@ set(JANSSON_WITHOUT_TESTS OFF)
 set(JANSSON_INSTALL OFF)
```

Comment:
Is this CMakeLists.txt needed at all? Let's get rid of it.

Reply:
Currently the `find_package_or_fallback_to_3party` macro has one conditional `add_subdirectory` statement in it.
And we need to add two directories (conditionally), one for the real 3party sources, and one for the generated code for the `jansson_config.h` and `jansson_private_config.h`.
How do you suggest to do this conditionally without chaining the two directories after each other, and without skipping the macro and writing a new conditional statement?

Reply:
Aren't these files needed only for XCode project? Can you remove them and try to (re)build? It would be great to generate XCode project later from CMake, then even this case will be solved.

Reply:
You were right. It indeed seems superfluous. Removed.